src/layers/SchematicLayer.ts

```diff
@@ -428,6 +507,288 @@ export class SchematicLayer<T extends SchematicData> extends PixiLayer<T> {
+            const end: [number, number] = [top[0], perfShapes[0].diameter];
+
+            canvasCtx.bezierCurveTo(...start, ...controlPoint1, ...middle);
+            canvasCtx.bezierCurveTo(...middle, ...controlPoint2, ...end);
+            canvasCtx.stroke();
+          }
+
+          for (let i = 0; i < amountOfSpikes; i++) {
+            const right: [number, number] = [i * spikeWidth + spikeWidth, canvas.height / 2];
+            const bottom: [number, number] = [right[0] - spikeWidth / 2, canvas.height - fracLineLenght - perfShapes[0].diameter];
+
+            canvasCtx.beginPath();
+
+            const start: [number, number] = [...bottom];
+            const controlPoint1: [number, number] = [bottom[0] - fracLineHalfWidth, canvas.height - fracLineLenght / 2 - perfShapes[0].diameter];
+            const middle: [number, number] = [bottom[0], canvas.height - fracLineLenght / 2 - perfShapes[0].diameter];
+            const controlPoint2: [number, number] = [bottom[0] + fracLineHalfWidth, canvas.height - fracLineLenght / 4 - perfShapes[0].diameter];
+            const end: [number, number] = [bottom[0], canvas.height - perfShapes[0].diameter];
+
+            canvasCtx.bezierCurveTo(...start, ...controlPoint1, ...middle);
+            canvasCtx.bezierCurveTo(...middle, ...controlPoint2, ...end);
+            canvasCtx.stroke();
+          }
+        },
+        // Yellow gravel. Makes perforations of type "Perforation" yellow if overlapping and perforation are open.
+        CasedHoleGravelPack: () => {
+          const size = DEFAULT_TEXTURE_SIZE * cementOptions.scalingFactor;
+          canvas.width = size / 2;
+          canvas.height = size;
+          canvasCtx.fillStyle = perforationColors.yellow;
+          canvasCtx.save();
+          canvasCtx.globalAlpha = packingOpacity;
+          canvasCtx.fillRect(0, 0, canvas.width, canvas.height);
+          canvasCtx.restore();
+        },
+        // Yellow gravel and fracturation lines.
+        // Makes perforations of type "Perforation" yellow if overlapping and perforation are open.
+        // If no perforation of type "perforation" are overlapping, there are no fracturation lines and no spikes.
+        // If a perforation of type "perforation" is overlapping,
+        // the fracturation lines extends from the tip of the perforation spikes into formation.
+        CasedHoleFracPack: () => {
+          const size = DEFAULT_TEXTURE_SIZE * cementOptions.scalingFactor;
+          canvas.width = size / 2;
+          canvas.height = size;
+          canvasCtx.fillStyle = perforationColors.yellow;
+          canvasCtx.save();
+          canvasCtx.globalAlpha = packingOpacity;
+          canvasCtx.fillRect(0, 0, canvas.width, canvas.height);
+          canvasCtx.restore();
+        },
+      },
+      perforation.subKind,
+    );
+
+    const notGonnaCacheThePerforationTextureForNow = Texture.from(canvas, { wrapMode: WRAP_MODES.CLAMP });
+    return notGonnaCacheThePerforationTextureForNow;
```

Comment:
Maybe figure out if we can cache it somehow 🤔

Reply:
No. Way too much work for too little gain. Let's do it if we notice bad perf.